pkg/intermediate/aggregate.go

```diff
@@ -482,14 +482,24 @@ func (a *AggregationProcess) aggregateRecords(incomingRecord, existingRecord ent
 		return nil
 	}
 	isLatest := false
+	var prevFlowEndSeconds, flowEndSecondsDiff uint32
 	if ieWithValue, _, exist := incomingRecord.GetInfoElementWithValue("flowEndSeconds"); exist {
 		if existingIeWithValue, _, exist2 := existingRecord.GetInfoElementWithValue("flowEndSeconds"); exist2 {
 			incomingVal := ieWithValue.GetUnsigned32Value()
 			existingVal := existingIeWithValue.GetUnsigned32Value()
-			if incomingVal > existingVal {
+			if incomingVal >= existingVal {
 				isLatest = true
 				existingIeWithValue.SetUnsigned32Value(incomingVal)
 			}
+			// Update the flowEndSecondsFromSource/DestinationNode fields, and compute
+			// the time difference between the incoming record and the last record.
+			if fillSrcStats {
+				prevFlowEndSeconds = a.updateFlowEndSecondsFromNodes(incomingRecord, existingRecord, "flowEndSecondsFromSourceNode", incomingVal)
+			}
+			if fillDstStats {
+				prevFlowEndSeconds = a.updateFlowEndSecondsFromNodes(incomingRecord, existingRecord, "flowEndSecondsFromDestinationNode", incomingVal)
+			}
+			flowEndSecondsDiff = incomingVal - prevFlowEndSeconds
```

Comment:
Is it possible that `incomingVal - prevFlowEndSeconds` is less than zero? I noticed that you are using `uint32` type for `flowEndSecondsDiff`, just worried about if there could be a corner case here.

Reply:
yes, i believe that's possible in the case when agent down and restart, replaying previous records. Although we currently don't have the replay mechanism as a reference.

I added the logic to ensure we only do the aggregation when the incoming record is the latest wrt its coming node. That's to say, if a record from source node doesn't have the latest `flowEndSecondsFromSourceNode`, we will skip the aggregation of this record, since it doesn't bring new information.